MapleServer2/Servers/Game/DungeonManager.cs

```diff
@@ -74,41 +74,41 @@ public DungeonSession GetDungeonSessionByInstanceId(int instanceId)
             return DungeonSessionList.FirstOrDefault(session => session.Value.DungeonInstanceId == instanceId).Value;
         }
 
-        public bool IsDungeonUsingFieldInstance(DungeonSession dungeonSession, FieldManager fieldManager, Player player) //alternatively this could be: IsFieldInstanceUsed in FieldManagerFactory
+        public bool IsDungeonUsingFieldInstance(FieldManager fieldManager, Player player) //alternatively this could be: IsFieldInstanceUsed in FieldManagerFactory
         {
-            if (dungeonSession == null) //is not null after entering dungeon via directory
+            DungeonSession currentDungeonSession = GetDungeonSessionByInstanceId(fieldManager.InstanceId);
+            if (currentDungeonSession == null) //is not null after entering dungeon via directory
             {
                 return false; //no dungeonsession -> the map is unused by dungeon
             }
             //fieldManager.MapId: left map that is to be destroyed
             //player.MapId: travel destination of the player
             //check map that is left: 
-            if (!dungeonSession.IsDungeonMap(fieldManager.MapId)) //left map is not dungeon map e.g. tria
+            if (!currentDungeonSession.IsDungeonSessionMap(fieldManager.MapId)) //left map is not dungeon map e.g. tria
             {
                 return false;
             }
-            else //left map is a dungeon map
+            //travel destination is a dungeon map: lobby to dungeon or dungeon to lobby
+            if (currentDungeonSession.IsDungeonSessionMap(player.MapId) && player.InstanceId == currentDungeonSession.DungeonInstanceId)
```

Comment:
Could break down this logic even further by taking each part, inverting, then returning false, then finally on line 96 you return true